lib/src/util/windows/user.cc

```diff
@@ -44,15 +46,24 @@ namespace facter { namespace util { namespace windows { namespace user {
 
     string home_dir()
     {
-        string home, alt;
-        if (environment::get("HOME", home)) {
-            return home;
-        } else if (environment::get("HOMEDRIVE", home) && environment::get("HOMEPATH", alt)) {
-            return home + alt;
-        } else if (environment::get("USERPROFILE", home)) {
-            return home;
+        HANDLE temp_token;
+        if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &temp_token)) {
+            LOG_DEBUG("OpenProcessToken call failed: %1%", system_error());
+            return {};
         }
-        return {};
+        scoped_resource<HANDLE> token(move(temp_token), CloseHandle);
+
+        DWORD pathLen;
+        GetUserProfileDirectoryW(token, nullptr, &pathLen);
```

Comment:
Although not specified in the msdn docs, the usual behavior is for the function to fail, and have GetLastError return ERROR_INSUFFICIENT_BUFFER and set the required buffer length in the in/out param. That way you can differentiate between "I need to call the API again with the new buffer length" vs "I should fail completely"